### PR TITLE
get a low level connection ID from reverse proxy server header

### PR DESCRIPTION
**Pull request by pavelmash**
TODO:
 - remove usage of SynCommons - temporary added for GetInt64
 - discuss changing of ConnectionID type to Int64 in all classes

### CODE REVIEW

File: SynCrtSock.pas [hunk 1]
```diff
@@ -6209,7 +6228,10 @@ constructor THttpServerResp.Create(aServerSock: THttpServerSocket; aServer: THtt
   finally
     LeaveCriticalSection(fServer.fProcessCS);
   end;
-  fConnectionID := fServer.NextConnectionID;
+  if (aServerSock.RemoteConnectionID > 0) then
```
Comment (achechulin) [inline SynCrtSock.pas]: I think should be:
`aServerSock.RemoteConnectionID` <> 0

File: SynCrtSock.pas [hunk 2]
```diff
@@ -3144,6 +3156,7 @@ implementation
 
 {$ifdef FPC}
 uses
+  SynCommons, // MPV - temporary for GetInt64
```
Comment (synopse) [inline SynCrtSock.pas]: you can use GetNextItemUInt64()

File: SynCrtSock.pas (hunk 1 above)
Comment (synopse) [reply SynCrtSock.pas]: +1

File: SynCrtSock.pas [hunk 3]
```diff
@@ -6594,6 +6616,17 @@ function THttpServerSocket.GetRequest(withBody: boolean=true): boolean;
         end else
           inc(H);
       end;
+      // remote connection ID
+      L := length(fServer.fRemoteConnIDHeaderUpper);
+      if L<>0 then  begin
+        H := pointer(Headers);
+        for i := 1 to length(Headers) do
+        if IdemPChar(H^,pointer(fServer.fRemoteConnIDHeaderUpper)) and (H^[L]=':') then begin
```
Comment (synopse) [inline SynCrtSock.pas]: why not just include the ':' in the fRemoteConnIDHeaderUpper ?

File: SynCrtSock.pas [hunk 4]
```diff
@@ -8629,6 +8663,19 @@ procedure THttpApiServer.Execute;
...
         Context.fUseSSL := Req^.pSslInfo<>nil;
         Context.fInHeaders := RetrieveHeaders(Req^,fRemoteIPHeaderUpper,RemoteIP);
+        // compute remote connection ID
+        L := length(fRemoteConnIDHeaderUpper);
+        if L<>0 then begin
+          P := Req^.Headers.pUnknownHeaders;
+          if P<>nil then
+          for i := 1 to Req^.Headers.UnknownHeaderCount do
+            if (P^.NameLength=L) and IdemPChar(P^.pName,Pointer(fRemoteConnIDHeaderUpper)) then begin
+              SetString(RemoteConn,p^.pRawValue,p^.RawValueLength);
```
Comment (synopse) [inline SynCrtSock.pas]: if you use GetNextItemUInt64(), local RemoteConn is not needed

File: SynCrtSock.pas (hunk 3 above)
Comment (pavelmash) [reply SynCrtSock.pas]: because we need  header without ':' in HTTP.SYS section

File: SynCrtSock.pas (hunk 4 above)
Comment (pavelmash) [reply SynCrtSock.pas]: We need local RemoteConn because `p^.pRawValue` not #0 ended (I adds a comment for this)

File: SynCrtSock.pas [hunk 5]
```diff
@@ -1901,8 +1913,8 @@   THttpServer = class(THttpServerGeneric)
     // - expects the port to be specified as string, e.g. '1234'; you can
     // optionally specify a server address to bind to, e.g. '1.2.3.4:1234'
     // - you can specify a number of threads to be initialized to handle
-    // incoming connections (default is 32, which may be sufficient for most
-    // cases, maximum is 256) - if you set 0, the thread pool will be disabled
+    // incoming connections. Default is 32, which may be sufficient for most
```
Comment (pavelmash) [inline SynCrtSock.pas]: I remove ( ) from comment because Lazarus parser not work correctly with ( ). Ctrl+Shift+Down not work (

**Pull request merged**